### PR TITLE
Refactor admin api routes

### LINKED ISSUE #230: [Auth] Split client and admin API routes
We have decided to split API routes for the Admin UI from the API routes for the Unleash clients. This will make it easier to maintain a stable API for the clients as well as adding more suited endpoints for the Admin UI.

This will also make it possible to protect the Unleash UI with Oauth or similar, and use other authentication mechanisms for the "read-only" clients (the will still be allowed to post back metrics), where client tokens as part of the `Authorization` will be more suited. 

This is the first task in to solving authentication which will be included of Unleash 3.x.

(#227 is solving this.)

### LINKED ISSUE #235: Drop support for v1.x clients in Unleash 3.x
We are commited to always be backward compatible for one major version between unleas-server and the unleash-clients. As we are working on Unleash 3.x we can now drop support for unleas v1.x clients which allows us to clean up:

- drop using legacyFeatureMapper (adds strategy together with strategies)
- drop `/features` path on root, now served on `/api/client/features`.

### PR DESCRIPTION
**Pull request by sveisvei**
* rewrite all routes to export a router
* clean up all api paths, closes #230 
* removes v.1 legacy routes, closes #235

Before merging:
* [x] Update API doc to reflect client / admin split
* [x] Manually verify that unleash-frontend is still working (after updating URI endpoints) https://github.com/Unleash/unleash-frontend/pull/80
* [x] Verify official clients that they still function
    * [x] node.js
    * [x] java
    * [x] go

After merge: 
* [x] release 3.0-alpha of unleash-frontend :white_check_mark: 
* [x] release 3.0-alpha of unleash :white_check_mark:

### CODE REVIEW

**Comment by ivarconr** [comment]
We also need to clean up e2e tests to be either client or admin.

**Comment by ivarconr** [comment]
inside "lib/routes" folder you have chosen this structure:

```
routes
 ├─admin-api
 ├─client-api
 ├─common
```

Would it be more concise to have it like this:

```
routes
 ├─api
   ├─admin
   ├─client
   ├─shared
```

File: lib/routes/admin-api/feature.js [hunk 1]
```diff
@@ -0,0 +1,206 @@
...
+    FEATURE_UPDATED,
+    FEATURE_ARCHIVED,
+} = require('../../event-type');
+const NameExistsError = require('../../error/name-exists-error');
+const NotFoundError = require('../../error/notfound-error');
+const ValidationError = require('../../error/validation-error.js');
+const validateRequest = require('../../error/validate-request');
+const extractUser = require('../../extract-user');
+
+const legacyFeatureMapper = require('../../data-helper/legacy-feature-mapper');
```
Comment (ivarconr) [inline lib/routes/admin-api/feature.js]: admin should not needed as this mapper is only there to support v1 clients.

File: lib/routes/admin-api/feature.js [hunk 2]
```diff
@@ -0,0 +1,206 @@
...
+module.exports.router = function(config) {
+    const { featureToggleStore, eventStore } = config.stores;
+    const router = Router(); // eslint-disable-line new-cap
+
+    router.get('/', commonFeatureToggle.router(config));
+
+    router.get('/:featureName', (req, res) => {
+        featureToggleStore
+            .getFeature(req.params.featureName)
+            .then(legacyFeatureMapper.addOldFields)
```
Comment (ivarconr) [inline lib/routes/admin-api/feature.js]: remove

File: lib/routes/common/feature.js [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+const { Router } = require('express');
+const version = 1;
+
+const legacyFeatureMapper = require('../../data-helper/legacy-feature-mapper');
+
+exports.router = config => {
+    const { featureToggleStore } = config.stores;
+    const router = Router(); // eslint-disable-line new-cap
+
+    router.get('/', (req, res) => {
```
Comment (ivarconr) [inline lib/routes/common/feature.js]: I feel it is cheaper to duplicate this than to share it. E.g. we don't need the legacy mapper in admin, because this is only needed to support old clients.

**Comment by ivarconr** [comment]
Generally this PR looks promising. I can help out doing the "manual" verifications when all cleanups is done.

File: lib/routes/client-api/feature.js [hunk 4]
```diff
@@ -0,0 +1,12 @@
+'use strict';
+
+const { Router } = require('express');
+const commonFeatureToggle = require('../common/feature.js');
+
+exports.router = config => {
+    const router = Router(); // eslint-disable-line new-cap
+
+    router.get('/', commonFeatureToggle.router(config));
```
Comment (ivarconr) [inline lib/routes/client-api/feature.js]: Actually this will not be sufficient, we have "non-official" clients (ha-proxy-integration) fetching the actual toggles directly (`/features/:toggleName`)

File: lib/routes/admin-api/feature.js (hunk 1 above)
Comment (sveisvei) [inline lib/routes/admin-api/feature.js]: Might as well :) 👍

File: lib/routes/common/feature.js (hunk 3 above)
Comment (sveisvei) [inline lib/routes/common/feature.js]: Indeed.

File: .eslintrc [hunk 5]
```diff
@@ -1,12 +1,22 @@
 {
     "extends": [
         "finn",
-        "finn/node"
+        "finn/node",
+        "finn-prettier"
```
Comment (SimenB) [inline .eslintrc]: should be in its own PR to reduce noise here, right?

Comment (ivarconr) [inline .eslintrc]: I general yes that would be preferable, but I don't want to burden @sveisvei with to much extra work to separate it in it's own PR...

Comment (SimenB) [inline .eslintrc]: I can do the rebase, no problem.

**Comment by sveisvei** [comment]
@ivarconr please review 💯

File: .eslintrc (hunk 5 above)
Comment (sveisvei) [inline .eslintrc]: I did squash  them into own commits -so its better now.

Comment (SimenB) [inline .eslintrc]: Yup, easier to review now! 🙂
